File: sorting_algo/python3codes/quick_sort.py
def partition(arr, l, r):
    pivot = arr[r]

    index = l 
    for i in range(l, r-1):
        if arr[i]<pivot:
            arr[i], arr[index] = arr[index], arr[i]
            index+=1 
    arr[r] = arr[index] 
    arr[index] = pivot 

    return index

def quickSort(arr, l, r):
    if l>=r: return 

    partition_index = partition(arr, l, r)
    quickSort(arr, l, partition_index-1)
    quickSort(arr, partition_index+1, r)

arr = [3, 7, 1, 33, -12, 4, 12, 99, -6]
print(arr)
quickSort(arr, 0, len(arr)-1)
print('quickSort')
print(arr)

File: sorting_algo/quick_sort.cpp
#include<iostream>
using namespace std;

void disp(int arr[], int n)
{
    cout<<endl;
    for(int i=0; i<n; i++)
        cout<<arr[i]<<"  ";
}

int partition(int arr[], int l, int r)
{
    int pivot = arr[r];

    int index=l;
    for(int i=l; i<r; i++)
    {
        if(arr[i]<pivot)
        {
            int temp = arr[index];
            arr[index] = arr[i];
            arr[i] = temp;
            index++;
        }
    }
    arr[r] = arr[index];
    arr[index] = pivot;
    return index;
}

void quickSort(int arr[], int l, int r)
{
    if(l>=r) return;
    int partition_index = partition(arr, l, r);
    quickSort(arr, l, partition_index-1);
    quickSort(arr, partition_index+1, r);
}

int main()
{
    int arr[] = {6, 23, 1, 34, -12, 12};
    int n = sizeof(arr)/sizeof(arr[0]);
    disp(arr, n);
    //call sort func 
    quickSort(arr, 0, n-1);
    disp(arr, n);

    return 0;
}